src/solver/types.jl

```diff
@@ -375,123 +375,58 @@ struct AtmosConfig{FT, TD, PA, C}
-
-    parsed_args = merge(
-        parsed_args_defaults,
-        parsed_args_perf_target,
-        parsed_args_prescribed,
-    )
-    return parsed_args
-end
-
-
 """
-    AtmosCoveragePerfParsedArgs()
-
-Define an atmos config for performance runs, and allows
-options to be overridden in several ways. In short the
-precedence for defining `parsed_args` is
-
-    - Highest precedence: args defined in `ARGS`
-    - Mid     precedence: args defined in `parsed_args_perf_target` (below)
-    - Lowest  precedence: args defined in `cli_defaults(s)`
-
-# Example usage:
-
-Launch with `julia --project=perf/`
-
-```julia
-import ClimaAtmos as CA
-import Random
-Random.seed!(1234)
-parsed_args = CA.AtmosCoveragePerfParsedArgs(;moist="dry")
-config = CA.AtmosConfig(;parsed_args)
-```
+    AtmosCoveragePerfConfig()
+    AtmosCoveragePerfConfig(s)
+Creates a model configuration for many performance tests.
+Creates a config from the following in order of top priority to last:
+1. Configuration from the given config file
+2. Default perf configuration
+3. Target job configuration
+4. Default configuration
 """
-function AtmosCoveragePerfParsedArgs(s = argparse_settings())
-    @info "Using coverage performance parameters + provided CL arguments."
-    parsed_args_defaults = cli_defaults(s)
-    dict = parsed_args_per_job_id(; filter_name = "driver.jl")
-
-    # Start with performance target, and override anything provided in ARGS
-    parsed_args_prescribed = parsed_args_from_ARGS(ARGS)
-
-    target_job = get(parsed_args_prescribed, "target_job", nothing)
-    parsed_args_perf_target = isnothing(target_job) ? Dict() : dict[target_job]
-
-    parsed_args_perf_target["forcing"] = "held_suarez"
-    parsed_args_perf_target["vert_diff"] = true
-    parsed_args_perf_target["surface_setup"] = "DefaultExchangeCoefficients"
-    parsed_args_perf_target["moist"] = "equil"
-    parsed_args_perf_target["rad"] = "allskywithclear"
-    parsed_args_perf_target["precip_model"] = "0M"
-    parsed_args_perf_target["dt"] = "1secs"
-    parsed_args_perf_target["t_end"] = "10secs"
-    parsed_args_perf_target["dt_save_to_sol"] = Inf
-    parsed_args_perf_target["z_elem"] = 25
-    parsed_args_perf_target["h_elem"] = 12
-
-    parsed_args = merge(
-        parsed_args_defaults,
-        parsed_args_perf_target,
-        parsed_args_prescribed,
-    )
-    return parsed_args
-end
-
-AtmosCoveragePerfConfig(s = argparse_settings()) =
-    AtmosConfig(s; parsed_args = AtmosCoveragePerfParsedArgs(s))
-
-function AtmosConfigArgs(
-    s = argparse_settings();
-    args = String[],
-    parsed_args = parse_commandline(args, s),
-    comms_ctx = get_comms_context(parsed_args),
-)
-    @info "Running ClimaAtmos with default argparse settings + $args"
-    return AtmosConfig(s; parsed_args, comms_ctx)
+function AtmosCoveragePerfConfig(s = argparse_settings())
+    parsed_args = parse_commandline(s)
+    config_dict = merge(YAML.load_file.(parsed_args["config_file"])...)
```

Comment:
This is a bit confusing, is `config_file` a list of files? This is broadcasting `load_file` over it and then merging the dicts

Reply:
I think `config_file` can contain either be a single filepath or a list of filepaths. I thought that might be more flexible, but now it seems like it should be restricted to just one.